main/src/library/Int16.flix

```diff
@@ -271,6 +271,18 @@ namespace Int16 {
     @Time(1) @Space(1)
     pub def flipBit(x: Int16, position: Int32): Int16 = x ^^^ (1i16 <<< position)
 
+    ///
+    /// Returns the integer binary logarithm of `x`.
+    ///
+    @Time(1) @Space(1)
+    pub def log2(x: Int16): Int32 = highestOneBitPosition(x)
```

Comment:
It seems that other methods like `Int16.highestOneBitPosition` return an `Int32` value, so I make both `log2` and `factorial` returning `Int32` too.